### PR TITLE
Quartz Solar v0.5.2 – Staging

### PR DESCRIPTION
**Pull request by braddf**
# Pull Request

## Description

Includes a bundle of UI/UX improvements and features for the Quartz Solar UK UI.
- 🕑 N-hour forecast, replacing 4hr forecast functionality with a dropdown to switch forecast horizon;
- 📈  Chart legend – simplify + add toggles;
- 📈 Fix responsiveness for different screen sizes large and small (within reason, not beautiful on mobile but it should roughly work now at least);
- 📈 National full height charts when GSP not selected;
- 📈 Disable untidy chart animations;
- 🔽 Tidy Delta view table/layout;
- 0️⃣  Sites empty / welcome state;
- 🗺️ PV Forecast map tooltip, showing actuals/capacities/forecast values, and dynamic to the map mode;
- ⭐ Correct/nicer page title in browser tab

Fixes #

## How Has This Been Tested?

- [x] Cypress tests running (and caught an issue with the profile dropdown I had inadvertently caused, before I had committed! Good job, tests 🙌 )
- [x] Locally and on Dev

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [x] Yes, some tweaks to map functionality and all seems okay to me

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: apps/nowcasting-app/components/Toggle.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { FC } from "react";
```
Comment (peterdudfield) [inline apps/nowcasting-app/components/Toggle.tsx:1]: Is it good pratice to put a sentence at the very start of the file, saying roughly what this file does?

File: apps/nowcasting-app/components/charts/gsp-pv-remix-chart/use-get-gsp-data.ts [hunk 2]
```diff
@@ -25,7 +28,9 @@ const useGetGspData = (gspId: number) => {
```
Comment (peterdudfield) [inline apps/nowcasting-app/components/charts/gsp-pv-remix-chart/use-get-gsp-data.ts:28]: Is it more readable, if you declare. a variblae called `nMinuteForecast = nhourForecast * 60` and then use this in the url below?

File: apps/nowcasting-app/constant.ts [hunk 3]
```diff
@@ -102,3 +112,5 @@ export enum DELTA_BUCKET {
   POS4 = 100
 }
 export const getDeltaBucketKeys = () => Object.keys(DELTA_BUCKET).filter((k) => isNaN(Number(k)));
+
+export const NHR_FORECAST_OPTIONS = [1, 2, 4, 8];
```
Comment (peterdudfield) [inline apps/nowcasting-app/constant.ts:116]: There's a few place where `Nhr` and NHour` get used, Should we try to set them as the same? Perhaps use `NHour` as its slightly more explicit.

**Review by peterdudfield — CHANGES_REQUESTED** [review]
Looks great, and loads of things in it. In general it feels a big PR. 

Ive put a few minor comments them, perhaps the biggest being NHour or NHr. Some of the of the comments are questions, so feel free to dismiss them as you see fit

File: apps/nowcasting-app/components/Toggle.tsx (hunk 1 above)
Comment (braddf) [reply apps/nowcasting-app/components/Toggle.tsx:1]: I've not really seen that in JS/TS world, usually it's a combo of Types and component naming convention to describe what to expect from the component/file

File: apps/nowcasting-app/components/charts/gsp-pv-remix-chart/use-get-gsp-data.ts (hunk 2 above)
Comment (braddf) [reply apps/nowcasting-app/components/charts/gsp-pv-remix-chart/use-get-gsp-data.ts:28]: Can do 👌🏻

File: apps/nowcasting-app/constant.ts (hunk 3 above)
Comment (braddf) [reply apps/nowcasting-app/constant.ts]: Have refactored, hopefully almost all out now – left ENV vars just for the moment as can add later

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**